subprojects/core-api/src/main/java/org/gradle/api/artifacts/repositories/AuthenticationSupported.java

```diff
@@ -96,6 +97,33 @@
      */
     <T extends Credentials> void credentials(Class<T> credentialsType, Action<? super T> action);
 
+    /**
+     * Configures the credentials for this repository that will be provided by the build.
+     * <p>
+     * Credentials will be provided from Gradle properties based on the repository name.
+     * If credentials for this repository can not be resolved and the repository will be used in the current build, then the build will fail to start and point to the missing configuration.
+     * <pre class='autoTested'>
+     * repositories {
+     *     maven {
+     *         url "${url}"
+     *         credentials(PasswordCredentials)
+     *     }
+     * }
+     * </pre>
+     * <p>
+     * The following credential types are currently supported for the {@code credentialsType} argument:
+     * <ul>
+     * <li>{@link org.gradle.api.credentials.PasswordCredentials}</li>
```

Comment:
Stumbled upon this and realized it is linked with #13107.
 
I think that is a nice usability change. 👍 

However this class was using the `org.gradle.api.artifacts.repositories.PasswordCredentials` type instead by default.
I wonder why it exists in parallel of `org.gradle.api.credentials.PasswordCredentials` and wonder if we should not also take this as an opportunity to clean that up.

Reply:
Looks like `org.gradle.api.artifacts.repositories.PasswordCredentials` was where `PasswordCredentials` interface lived originally before all the `Credentials` APIs were moved to `org.gradle.api.credentials`. I will update the Javadoc to suggest `org.gradle.api.credentials.PasswordCredentials` instead.

We might also want to then deprecate `org.gradle.api.artifacts.repositories.PasswordCredentials` for removal in Gradle 7.